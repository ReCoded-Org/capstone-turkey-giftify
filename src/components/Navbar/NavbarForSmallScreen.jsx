import React, { useState, useEffect, useRef } from "react";
import { Link } from "react-router-dom";
import { List } from "react-bootstrap-icons";
import { Button } from "react-bootstrap";
import { smallScreenNavbarData } from "./../../data/Navbar/smallScreenNavbarData.js";

const NavbarForSmallScreen = () => {
  const [sidebar, setSidebar] = useState(false);

  const showSidebar = () => setSidebar(!sidebar);

  const boxRef = useRef(null);

  useEffect(() => {
    const handleClickOutside = (event) => {
      if (boxRef && boxRef.current.contains(event.target)) {
        setSidebar(true);
      } else {
        setSidebar(false);
      }
    };

    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, []);

  return (
    <>
      <div className="navbarMobile">
        <Link to="#" className="menuBars">
          <List onClick={showSidebar} />
        </Link>
        <Link to="/" className="navbarMobileLogo">
          LOGO
        </Link>
      </div>
      <nav ref={boxRef} className={sidebar ? "navMenu active" : "navMenu"}>
        <ul className="navMenuItems">
          <li className="navbarToggle" onClick={showSidebar}>
            <Link to="#" className="menuBars">
              <List />
            </Link>
          </li>

          <li className="navLi navLiLoginSign" onClick={showSidebar}>
            <Link to="/logIn">
              <Button
                variant="primary"
                style={{ borderRadius: "0.75rem", marginBottom: "5%" }}
              >
                Log In
              </Button>
            </Link>
          </li>
          <li className="navLi navLiLoginSign" onClick={showSidebar}>
            <Link to="/signUp">
              <Button
                variant="primary"
                style={{ borderRadius: "0.75rem", marginBottom: "5%" }}
              >
                Sign Up
              </Button>
            </Link>
          </li>
          {smallScreenNavbarData.map((info) => (
            <li className="navLi" onClick={showSidebar} key={info.id}>
              <Link to={info.target} className="navText">
                {info.component} &nbsp; {info.name}
              </Link>
            </li>
          ))}
        </ul>
      </nav>
    </>
  );
};

export default NavbarForSmallScreen;
