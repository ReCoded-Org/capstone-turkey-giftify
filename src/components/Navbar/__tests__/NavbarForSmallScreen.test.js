import React from "react";
import NavbarForSmallScreen from "./../NavbarForSmallScreen";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("matches snapshot", () => {
  const tree = renderer.create(<NavbarForSmallScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
