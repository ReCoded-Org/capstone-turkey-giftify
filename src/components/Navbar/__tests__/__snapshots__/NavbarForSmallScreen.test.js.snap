// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarForSmallScreen rendered correctly 1`] = `
Array [
  <div
    className="navbarMobile"
  >
    <a
      className="menuBars"
      href="/"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2.5 12a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 0 1H3a.5.5 0 0 1-.5-.5zm0-4a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 0 1H3a.5.5 0 0 1-.5-.5zm0-4a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 0 1H3a.5.5 0 0 1-.5-.5z"
          fillRule="evenodd"
        />
      </svg>
    </a>
    <a
      className="navbarMobileLogo"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Logo"
        className=""
        height="70px"
        src="navBarLogo.png"
        width="180px"
      />
    </a>
  </div>,
  <nav
    className="navMenu"
  >
    <ul
      className="navMenuItems"
    >
      <li
        className="navbarToggle"
        onClick={[Function]}
      >
        <a
          className="menuBars"
          href="/"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2.5 12a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 0 1H3a.5.5 0 0 1-.5-.5zm0-4a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 0 1H3a.5.5 0 0 1-.5-.5zm0-4a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 0 1H3a.5.5 0 0 1-.5-.5z"
              fillRule="evenodd"
            />
          </svg>
        </a>
      </li>
      <li
        className="navLi navLiLoginSign"
        onClick={[Function]}
      >
        <a
          href="/logIn"
          onClick={[Function]}
        >
          <button
            className="btn btn-primary"
            disabled={false}
            style={
              Object {
                "borderRadius": "0.75rem",
                "marginBottom": "5%",
              }
            }
            type="button"
          >
            Log In
          </button>
        </a>
      </li>
      <li
        className="navLi navLiLoginSign"
        onClick={[Function]}
      >
        <a
          href="/signUp"
          onClick={[Function]}
        >
          <button
            className="btn btn-primary"
            disabled={false}
            style={
              Object {
                "borderRadius": "0.75rem",
                "marginBottom": "5%",
              }
            }
            type="button"
          >
            Sign Up
          </button>
        </a>
      </li>
      <li
        className="navLi"
        onClick={[Function]}
      >
        <a
          className="navText"
          href="/"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.354 1.146a.5.5 0 0 0-.708 0l-6 6A.5.5 0 0 0 1.5 7.5v7a.5.5 0 0 0 .5.5h4.5a.5.5 0 0 0 .5-.5v-4h2v4a.5.5 0 0 0 .5.5H14a.5.5 0 0 0 .5-.5v-7a.5.5 0 0 0-.146-.354L13 5.793V2.5a.5.5 0 0 0-.5-.5h-1a.5.5 0 0 0-.5.5v1.293L8.354 1.146zM2.5 14V7.707l5.5-5.5 5.5 5.5V14H10v-4a.5.5 0 0 0-.5-.5h-3a.5.5 0 0 0-.5.5v4H2.5z"
            />
          </svg>
             
          Home
        </a>
      </li>
      <li
        className="navLi"
        onClick={[Function]}
      >
        <a
          className="navText"
          href="/search"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
            />
          </svg>
             
          Search
        </a>
      </li>
      <li
        className="navLi"
        onClick={[Function]}
      >
        <a
          className="navText"
          href="/requests"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 2.5a2.5 2.5 0 0 1 5 0 2.5 2.5 0 0 1 5 0v.006c0 .07 0 .27-.038.494H15a1 1 0 0 1 1 1v2a1 1 0 0 1-1 1v7.5a1.5 1.5 0 0 1-1.5 1.5h-11A1.5 1.5 0 0 1 1 14.5V7a1 1 0 0 1-1-1V4a1 1 0 0 1 1-1h2.038A2.968 2.968 0 0 1 3 2.506V2.5zm1.068.5H7v-.5a1.5 1.5 0 1 0-3 0c0 .085.002.274.045.43a.522.522 0 0 0 .023.07zM9 3h2.932a.56.56 0 0 0 .023-.07c.043-.156.045-.345.045-.43a1.5 1.5 0 0 0-3 0V3zM1 4v2h6V4H1zm8 0v2h6V4H9zm5 3H9v8h4.5a.5.5 0 0 0 .5-.5V7zm-7 8V7H2v7.5a.5.5 0 0 0 .5.5H7z"
            />
          </svg>
             
          Requests
        </a>
      </li>
      <li
        className="navLi"
        onClick={[Function]}
      >
        <a
          className="navText"
          href="/donations"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m8 2.748-.717-.737C5.6.281 2.514.878 1.4 3.053c-.523 1.023-.641 2.5.314 4.385.92 1.815 2.834 3.989 6.286 6.357 3.452-2.368 5.365-4.542 6.286-6.357.955-1.886.838-3.362.314-4.385C13.486.878 10.4.28 8.717 2.01L8 2.748zM8 15C-7.333 4.868 3.279-3.04 7.824 1.143c.06.055.119.112.176.171a3.12 3.12 0 0 1 .176-.17C12.72-3.042 23.333 4.867 8 15z"
            />
          </svg>
             
          Donations
        </a>
      </li>
      <li
        className="navLi"
        onClick={[Function]}
      >
        <a
          className="navText"
          href="/about"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
            />
            <path
              d="m8.93 6.588-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588zM9 4.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0z"
            />
          </svg>
             
          About
        </a>
      </li>
      <li
        className="navLi"
        onClick={[Function]}
      >
        <a
          className="navText"
          href="/contactUs"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 4a2 2 0 0 1 2-2h12a2 2 0 0 1 2 2v8a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2V4zm2-1a1 1 0 0 0-1 1v.217l7 4.2 7-4.2V4a1 1 0 0 0-1-1H2zm13 2.383-4.758 2.855L15 11.114v-5.73zm-.034 6.878L9.271 8.82 8 9.583 6.728 8.82l-5.694 3.44A1 1 0 0 0 2 13h12a1 1 0 0 0 .966-.739zM1 11.114l4.758-2.876L1 5.383v5.73z"
            />
          </svg>
             
          Contact Us
        </a>
      </li>
      <li
        className="navLi navLiLang"
      >
        <svg
          fill="currentColor"
          height="1em"
          style={
            Object {
              "color": "black",
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.166 8.94c-.524 1.062-1.234 2.12-1.96 3.07A31.493 31.493 0 0 1 8 14.58a31.481 31.481 0 0 1-2.206-2.57c-.726-.95-1.436-2.008-1.96-3.07C3.304 7.867 3 6.862 3 6a5 5 0 0 1 10 0c0 .862-.305 1.867-.834 2.94zM8 16s6-5.686 6-10A6 6 0 0 0 2 6c0 4.314 6 10 6 10z"
          />
          <path
            d="M8 8a2 2 0 1 1 0-4 2 2 0 0 1 0 4zm0 1a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
          />
        </svg>
        <div
          className="dropdownBasicButton dropdown"
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Language
          </button>
        </div>
      </li>
    </ul>
  </nav>,
]
`;
